engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/join/FullOuterJoinTransformer.scala

```diff
@@ -0,0 +1,162 @@
+      NextParameters(
+        List(KeyParam, AggregatorParam, AggregateByParam, WindowLengthParam).map(_.parameter), errors_names ++ errors_key)
+
+    case TransformationStep(
+    (`KeyParamName`, _) ::
+      (`AggregatorParamName`, DefinedEagerBranchParameter(aggregatorByBranchId: Map[String, Aggregator]@unchecked, _)) ::
+      (`AggregateByParamName`, aggregateByParam) ::
+      (`WindowLengthParamName`, _) ::
+      Nil, _) =>
+      val outName = OutputVariableNameDependency.extract(dependencies)
+      val mainCtx = contexts.headOption.map(_._2.clearVariables).getOrElse(ValidationContext())
+
+      val validatedOutputType: ValidatedNel[ProcessCompilationError, TypingResult] = aggregateByParam match {
+        case DefinedLazyBranchParameter(aggregateByByBranchId) =>
+          val validatedAggregatorReturnTypes = aggregatorByBranchId.map{case (id, agg) =>
+            agg.computeOutputType(aggregateByByBranchId(id).returnType)
+              .leftMap(x => {
+                val branchParamId = ParameterNaming.getNameForBranchParameter(AggregateByParam.parameter, id)
+                NonEmptyList(CustomNodeError(x, Some(branchParamId)), Nil)})
+              .map(id -> _)
+          }.toList.sequence.map(_.toMap)
+          validatedAggregatorReturnTypes.map(outputTypeByBranchId => {
+            val outputTypes = outputTypeByBranchId.toList
+              .map{case (k, v) => ContextTransformation.sanitizeBranchName(k) -> v}
+              .++(List(KeyFieldName -> Typed.typedClass[String]))
+            TypedObjectTypingResult(outputTypes, Typed.typedClass[java.util.Map[String, AnyRef]])
+          })
+
+        case _ => Validated.validNel(Unknown)
+      }
+
+      FinalResults.forValidation(mainCtx, validatedOutputType.swap.map(_.toList).getOrElse(List()))(
+        _.withVariable(OutputVar.customNode(outName), validatedOutputType.getOrElse(Unknown)))
+  }
+
+  override def implementation(params: Map[String, Any], dependencies: List[NodeDependencyValue], finalState: Option[State]): FlinkCustomJoinTransformation = {
+    val keyByBranchId: Map[String, LazyParameter[CharSequence]] = KeyParam.extractValue(params)
+    val aggregatorByBranchId: Map[String, Aggregator] = AggregatorParam.extractValue(params)
+    val aggregateByByBranchId: Map[String, LazyParameter[AnyRef]] = AggregateByParam.extractValue(params)
+    val window: Duration = WindowLengthParam.extractValue(params)
+
+    val aggregator: Aggregator = new MapAggregator(
+      aggregatorByBranchId.mapValues(new OptionAggregator(_).asInstanceOf[Aggregator]).asJava)
+
+    val baseElement: Map[String, AnyRef] = keyByBranchId.keySet.map(_ -> None).toMap
+
+    (inputs: Map[String, DataStream[Context]], context: FlinkCustomNodeContext) => {
+      val keyedStreams = inputs.map {
+        case (id, stream) =>
+          stream.flatMap(new StringKeyedValueMapper(context, keyByBranchId(id), aggregateByByBranchId(id)))
+            .map(_.map(_.mapValue(x => {
```

Comment:
can u write it that way:
```
.map(_.map(_.mapValue { x =>
...
```
We've got one brace less and it will be easier to read